### PR TITLE
first step towards simplifying protocol loading

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR introduces functionality for protocols to be loaded directly from a directory in one line via `Protocol.from_dir(protocol_dir_path)`.

## Fixes

- #834

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: docs/build-aea-programmatically.md [hunk 1]
```diff
@@ -86,19 +82,8 @@ Then we pass all of this into the AEA constructor to create our AEA.
...
-    )
-    default_protocol = Protocol(
-        PublicId.from_str("fetchai/default:0.1.0"),
-        DefaultSerializer(),
-        default_protocol_configuration,
-    )
-    resources.protocol_registry.register(
-        PublicId.from_str("fetchai/default:0.1.0"), default_protocol
-    )
+    default_protocol = Protocol.from_dir(os.path.join(AEA_DIR, "protocols", "default"))
```
Comment (DavidMinarsch) [inline docs/build-aea-programmatically.md]: This is a much more succinct approach and mirrors how we load skills.

Comment (DavidMinarsch) [reply docs/build-aea-programmatically.md]: Ideally, we'd also do the same for connection loading

File: tests/test_protocols/test_base.py [hunk 2]
```diff
@@ -107,3 +111,32 @@ def test_protocols_config(self):
             config=ProtocolConfig(),
         )
         assert protocol.config is not None
+
+
+class TestProtocolFromDir:
+    """Test the 'Protocol.from_dir' method."""
+
+    @classmethod
+    def setup_class(cls):
```
Comment (marcofavorito) [inline tests/test_protocols/test_base.py]: I think we can remove the creation of temporary directory. But it is OK for future extensions.

Comment (DavidMinarsch) [reply tests/test_protocols/test_base.py]: that's why I added it ;)..not because we need it lol

**Pull request merged**